### PR TITLE
Store the 'start' date rather than recreating on browser restart.

### PR DESCRIPTION
**Pull request by mikewest**
This patch also renames some of the variables in 'maybeShowNotification' for
(hopefully) clarity.

R=@msramek

### CODE REVIEW

File: chrome/sunset.js [hunk 1]
```diff
@@ -68,16 +57,21 @@ Sunset.article_url_ = "https://support.google.com/chrome_webstore/?p=keep_my_opt
...
- * Shows a notification if enough time has passed since the previous one
- * was not shown.
+ * Shows a notification if enough time has passed since the previous one.
+ *
  * @private
  */
 Sunset.maybeShowNotification_ = function() {
   chrome.storage.sync.get(null, function(data) {
+    // Populate the starting date with today's date if it isn't already set.
+    var starting_date = data.start ? new Date(data.start) : new Date();
```
Comment (msramek) [inline chrome/sunset.js]: If data.start is a random string, starting_date can be invalid. I would test (new Date(data.start)).getTime() instead. This is true iff the date object contains a valid date.

File: chrome/sunset.js [hunk 2]
```diff
@@ -68,16 +57,21 @@ Sunset.article_url_ = "https://support.google.com/chrome_webstore/?p=keep_my_opt
...
+    if (!data.start)
```
Comment (msramek) [inline chrome/sunset.js]: Similarly here.

File: chrome/sunset.js [hunk 3]
```diff
@@ -88,17 +82,17 @@ Sunset.maybeShowNotification_ = function() {
     // Trigger notification if we have reached the date suggested
     // by the timeline and if enough time has passed since the previous
     // one was shown. These checks are not required for the first notification.
-    var today = new Date();
+    var now = new Date();
```
Comment (msramek) [inline chrome/sunset.js]: Nit: We can also have "today" if we setHours(0, 0, 0, 0), although it probably doesn't matter much if the timer goes off after midnight or later during the day.

**Comment by mikewest** [comment]
Take another look?

File: chrome/sunset.js [hunk 4]
```diff
@@ -43,9 +34,7 @@ Sunset.start_ = new Date();  // TBD
  */
 Sunset.timeline_offsets_ = [
   0,
-  14,
-  28,
-  29
+  14
```
Comment (msramek) [inline chrome/sunset.js]: 15 should be listed here as well, as the last date here represents the uninstallation date.

**Comment by msramek** [comment]
LGTM.

**Pull request merged**